### DISCUSSION TITLE
What is the datalog engine used internally in Flix?

### DISCUSSION THREAD

**Discussion by aryx** [flix/flix#5467] [General]
How good is the datalog engine used in Flix? How does it compare to Souffle or LogiBlox? Can we write
pointer analysis in Flix and get the same speed we would get with Souffle? Did you use Flix for writing static analyzers
and did it scale to codebase with millions LOC?

**Comment by magnus-madsen** [comment]
The Datalog engine is decent; it uses the right algorithms and data structures, but it is nowhere near as scalable as Souffle which represent a multiple man-year effort. That said, if you have a large Flix program, we would be happy to work on performance improvements; once we have a good benchmarks.

**Reply by aryx** [reply]
I would not expect the Flix program to be large, but if I would use it for pointer analysis, I would expect the datalog facts to be very large.

**Comment by magnus-madsen** [comment]
@aryx If you are interested, we can discuss more in private (either on Gitter or on Zoom). 

We have been looking for someone interested in using Flix on a large scale and we would be happy to help them get started, if that is of interest.

**Reply by aryx** [reply]
We're just in the evaluation stage, looking at possible datalog engines for encoding some pointer and taint analysis.

**Reply by magnus-madsen** [reply]
No worries. Let's us know if you need help. 

Be sure to check out https://doc.flix.dev/ if you have not already.
